src/commands/createNewProject/validateFuncCliInstalled.ts

```diff
@@ -0,0 +1,107 @@
+
+export async function validateFuncCliInstalled(ui: IAzureUserInput, outputChannel: OutputChannel): Promise<void> {
+    await callWithTelemetryAndErrorHandling('azureFunctions.validateFuncCliInstalled', ext.reporter, undefined, async function (this: IActionContext): Promise<void> {
+        this.suppressErrorDisplay = true;
+        const settingKey: string = 'showFuncInstallation';
+        if (getFuncExtensionSetting<boolean>(settingKey)) {
+            if (!(await funcCliInstalled()) && (await brewOrNpmInstalled()) && process.platform !== Platform.Linux) {
+                // https://github.com/Microsoft/vscode-azurefunctions/issues/311
+                const input: MessageItem = await ui.showWarningMessage(localize('installFuncCli', 'You need Azure Functions Core Tools CLI to locally debug your functions.  Run {0} now?', getCommandForPlatform()), DialogResponses.yes, DialogResponses.dontWarnAgain, DialogResponses.skipForNow);
+                if (input === DialogResponses.yes) {
+                    await attemptToInstallLatestFunctionRuntime(ui, outputChannel);
+                } else if (input === DialogResponses.dontWarnAgain) {
+                    await updateGlobalSetting(settingKey, false);
+                }
+            }
+        }
+    });
+}
+
+async function attemptToInstallLatestFunctionRuntime(ui: IAzureUserInput, outputChannel: OutputChannel): Promise<void> {
+    try {
+        switch (process.platform) {
+            case Platform.Windows:
+                const v1: MessageItem = { title: 'v1' };
+                const v2: MessageItem = { title: 'v2' };
+                const winput: MessageItem = await ui.showWarningMessage(localize('windowsVersion', 'Which version of the runtime do you want to install?'), v1, v2);
+                if (winput === v1) {
+                    await cpUtils.executeCommand(outputChannel, undefined, 'npm', 'install', '-g', 'azure-functions-core-tools');
+                } else if (winput === v2) {
+                    await cpUtils.executeCommand(outputChannel, undefined, 'npm', 'install', '-g', 'azure-functions-core-tools@core', '--unsafe-perm', 'true');
+                }
+                break;
+            case Platform.MacOS:
+                try {
+                    await cpUtils.executeCommand(outputChannel, undefined, 'brew', 'tap', 'azure/functions');
+                    await cpUtils.executeCommand(outputChannel, undefined, 'brew', 'install', 'azure-functions-core-tools');
+                } catch (error) {
+                    // if brew fails for whatever reason, fall back to npm
+                    await cpUtils.executeCommand(outputChannel, undefined, 'npm', 'install', '-g', 'azure-functions-core-tools@core', '--unsafe-perm', 'true');
+                }
+                break;
+            default:
+                break;
+        }
+    } catch (error) {
+        throw new Error('Installation of Azure Functions Core Tools CLI failed.');
```

Comment:
Remove this try/catch. We want to track the _actual_ error in telemetry and this will never be displayed to the user since suppressErrorDisplay is true